File: benchmarks/vm/03_hashmap.py
import time
t0 = time.monotonic()
m = {}
for i in range(100000):
    m[i] = i * 2
insert_ms = (time.monotonic() - t0) * 1000
t1 = time.monotonic()
total = 0
for j in range(100000):
    total += m.get(j, 0)
lookup_ms = (time.monotonic() - t1) * 1000
total_ms = int(insert_ms + lookup_ms)
print(f"total = {total}")
print(f"Insert: {int(insert_ms)} ms | Lookup: {int(lookup_ms)} ms")
print(f"BENCH_RESULT: {total_ms} ms")

File: benchmarks/vm/03_hashmap.rb
t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC)
m = {}
100_000.times { |i| m[i] = i * 2 }
insert_ms = (Process.clock_gettime(Process::CLOCK_MONOTONIC) - t0) * 1000
t1 = Process.clock_gettime(Process::CLOCK_MONOTONIC)
total = 0
100_000.times { |j| total += (m[j] || 0) }
lookup_ms = (Process.clock_gettime(Process::CLOCK_MONOTONIC) - t1) * 1000
total_ms = (insert_ms + lookup_ms).to_i
puts "total = #{total}"
puts "Insert: #{insert_ms.to_i} ms | Lookup: #{lookup_ms.to_i} ms"
puts "BENCH_RESULT: #{total_ms} ms"
